src/main/java/org/sonarlint/intellij/editor/SonarLintHighlighting.java

```diff
@@ -73,47 +68,46 @@ private void stopBlinking() {
     }
   }
 
-  /**
-   * Create highlighting with {@link UpdateHighlightersUtil}. It will manage internally the {@link RangeHighlighter}, and get
-   * it similarly to the way {@link com.intellij.codeHighlighting.Pass} do it.
-   * Tooltip will be displayed on mouse hover by {@link com.intellij.codeInsight.daemon.impl.DaemonListeners}.
-   * Creating the {@link HighlightInfo} with high severity will ensure that it will override most other highlighters.
-   * The alternative would be to get and manage directly {@link RangeHighlighter} with a {@link MarkupModel} from the
-   * document (or editors). This would allow to use a custom renderer, but we would have to manage tooltips by ourselves, separately.
-   *
-   * @see com.intellij.codeInsight.hint.HintManager
-   * @see com.intellij.codeInsight.hint.ShowParameterInfoHandler
-   * @see HintHint
-   * @see CustomHighlighterRenderer
-   * @see com.intellij.codeInsight.daemon.DaemonCodeAnalyzer
-   * @see com.intellij.codeInsight.highlighting.BraceHighlightingHandler
-   */
-  public void highlightFlowsWithHighlightersUtil(RangeMarker rangeMarker, @Nullable String message, List<LiveIssue.Flow> flows) {
-    stopBlinking();
-    HighlightInfo primaryInfo = createHighlight(rangeMarker, message);
-
-    List<HighlightInfo> infos = flows.stream()
-      .flatMap(f -> f.locations().stream()
+  public void highlightIssue(LiveIssue issue) {
+    RangeMarker issueRange = issue.getRange();
+    if (issueRange == null) {
+      return;
+    }
+    List<HighlightInfo> highlights = issue.flows().stream().findFirst()
```

Comment:
Remember the special case of issues having only flows with a single location (like cognitive complexity). Here all (single) locations of all flows should be highlighted.